### DISCUSSION TITLE
Convert Azi/Alt to Ra/Dec

### DISCUSSION THREAD

**Discussion by offsky** [cosinekitty/astronomy#129] [Q&A] [answered]
The `Horizon(date, observer, ra, dec, refraction)` function is very handy for converting ra/dec to azi/alt.  I have a need to do the reverse. I was looking for something like `Equator(body, date, observer, ofdate, aberration)` but instead of body, you can pass in any ra/dec. I can see in the docs that there are methods for transforming coordinates. I attempted to do this, but a round trip from Eq>Horiz>Eq is not getting me the same numbers, so I am doing something wrong. 

```
var mercury = Astronomy.Equator("Mercury", this.date, this.observer, true, true);
console.log(mercury); //RA/DEC
var horizon = Astronomy.Horizon(this.date, this.observer, mercury.ra, mercury.dec, true);
console.log(horizon); //AZI/ALT

//Converting from Horizon to Equator
var matrix = Astronomy.Rotation_HOR_EQD(this.date,this.observer);
var spherical = {lat:horizon.azimuth,lon:horizon.altitude,dist:mercury.dist};
var inVector = Astronomy.VectorFromHorizon(spherical,this.date,true);
var outVector = Astronomy.RotateVector(matrix,inVector);
var equator = Astronomy.EquatorFromVector(outVector);
console.log(equator); //This doesn't match RA/REC above.
```

I would really appreciate a nudge in the right direction.  One reason for doing this is to calculate the RA/DEC of the observer's zenith at a given time, among other points of interest.

**Comment by cosinekitty** [comment]
I think you are very close. There is one thing I see so far. The refraction parameter to `Horizon` and `VectorFromHorizon` is not supposed to be `true`. If you want refraction correction, you should pass the string `"normal"`. Otherwise pass `null`. I should change Astronomy Engine to throw an exception if someone passes an invalid value like `true`, so they notice something is wrong.

How different are the RA/DEC coordinates in `equator` to the original ones in `mercury`? Are they close or totally wrong?

**Comment by cosinekitty** [comment, marked as answer]
Oh, I think this is it:

```JS
var spherical = {lat:horizon.azimuth,lon:horizon.altitude,dist:mercury.dist};
```

It looks like `lat` and `lon` are backwards. Try this instead:

```JS
var spherical = new Astronomy.Spherical(horizon.altitude, horizon.azimuth, mercury.dist);
```

**Comment by offsky** [comment]
Doh! How did I get those reversed? I feel dumb now. 
Yes, that worked perfectly (along with changing true->"normal"). Thanks for the debugging assistance.

**Comment by cosinekitty** [comment]
It took me a while to spot it also. I have to think about it every time I do that... "is azimuth a longitude or a latitude?"

**Reply by gerben-b** [reply]
I am using this now to point my telescope to a chosen coordinate on my planetarium. This was very helpful, thanks.
